bin/assets/scripts/newform.js

```diff
@@ -2,6 +2,42 @@ const form = document.forms[0];
 const lang = form.lang;
 const langs = [...lang.options].slice(1).flatMap((option) => [option.value, option.textContent.toLowerCase()]);
 const code = form.code;
+const token = form.token;
+
+form.addEventListener('submit', async (event) => {
+  if (!token.value) {
+    return;
+  }
+  event.preventDefault();
+  const res = await fetch('/new?redirect=0', {
+    method: 'POST',
+    body: new FormData(form),
+  });
+  const content = await res.text();
+
+  if (!res.ok) {
+    const actual = document.documentElement.innerHTML;
+    history.pushState(undefined, undefined, '/new');
+    document.documentElement.innerHTML = content;
+
+    window.addEventListener('popstate', (event) => {
+      if (!event.state) {
+        document.documentElement.innerHTML = actual;
+      }
+    }, { once: true });
+    return;
+  }
```

Comment:
we should find a better solution

Reply:
Honnêtement, si on trouve rien et que tu arrives à le faire marcher, on merge ainsi, on est d'accord que c'est pas du beau code mais balek ya que nous qui le lirons.